Update to SvelteKit quickstart with auth helpers (#9754)
supabase/supabase: apps/reference/docs/guides/with-sveltekit.mdx

And one optional step is to update the CSS file `public/global.css` to make the app look nice.
You can find the full contents of this file [here](https://raw.githubusercontent.com/supabase/supabase/master/examples/user-management/svelte-user-management/src/app.css).

### Supabase Auth Helpers

SvelteKit is a highly versatile framework offering pre-rendering at build time (SSG), server-side rendering at request time (SSR), API routes, and more.

It can be challenging to authenticate your users in all these different environments, that's why we've created the Supabase Auth Helpers to make user management and data fetching within SvelteKit as easy as possible.

> Install the auth helpers for SvelteKit

Reviewer: ```suggestion
Install the auth helpers for SvelteKit:
```

```bash
npm install @supabase/auth-helpers-sveltekit
```

Update your `src/routes/+layout.svelte`:

```html title="src/routes/+layout.svelte"
<script lang="ts">
	import { supabaseClient } from '$lib/supabaseClient'
	import { invalidate } from '$app/navigation'
	import { onMount } from 'svelte'
	import './styles.css'

	onMount(() => {
		const {
			data: { subscription }
		} = supabaseClient.auth.onAuthStateChange(() => {
			invalidate('supabase:auth')
		})

		return () => {
			subscription.unsubscribe()
		}
	})
</script>

<div class="container" style="padding: 50px 0 100px 0">
	<slot />
</div>
```

> Create a new file called `src/routes/+layout.ts` which will take care of handling the session on the client-side.

Reviewer: ```suggestion
Create a new `src/routes/+layout.ts` file to handle the session on the client-side.
```

```ts title="src/routes/+layout.ts"
import type { LayoutLoad } from './$types';
import { getSupabase } from '@supabase/auth-helpers-sveltekit';

export const load: LayoutLoad = async (event) => {
  const { session } = await getSupabase(event);
  return { session };
};
```

> Create another new file called `src/routes/+layout.server.ts` which will take care of handling the session on the server-side.

Reviewer: ```suggestion
Create a new `src/routes/+layout.server.ts` file to handle the session on the server-side.
```

Let's set up a Svelte component to manage logins and sign ups. We'll use Magic Links, so users can sign in with their email without using passwords.

```html title="src/routes/Auth.svelte"
<script lang="ts">
	import { supabaseClient } from '$lib/supabaseClient'

	let loading = false
	let email: string

	const handleLogin = async () => {
		try {
			loading = true
			const { error } = await supabaseClient.auth.signInWithOtp({ email })
			if (error) throw error
			alert('Check your email for the login link!')
		} catch (error) {
			if (error instanceof Error) {
				alert(error.message)
			}
		} finally {
			loading = false
		}
	}
</script>

<form class="row flex-center flex" on:submit|preventDefault={handleLogin}>
	<div class="col-6 form-widget">
		<h1 class="header">Supabase + SvelteKit</h1>
		<p class="description">Sign in via magic link with your email below</p>
		<div>
			<input class="inputField" type="email" placeholder="Your email" bind:value={email} />
		</div>
		<div>
			<input
				type="submit"
				class="button block"
				value={loading ? 'Loading' : 'Send magic link'}
				disabled={loading}
			/>
		</div>
	</div>
</form>
```

### Account component

> After a user is signed in we can allow them to edit their profile details and manage their account.
> Let's create a new component for that called `Account.svelte`.

Reviewer: ```suggestion
After a user is signed in, they need to be able to edit their profile details and manage their account.
Create a new `Account.svelte` component to handle this functionality.
```
